src/main/scala/firrtl/FirrtlExecutionOptions.scala

```diff
@@ -0,0 +1,752 @@
+// See LICENSE for license details.
+
+package firrtl
+
+import firrtl.annotations.{
+  Annotation,
+  DeletedAnnotation,
+  SingleStringAnnotation,
+  NoTargetAnnotation,
+  LegacyAnnotation,
+  AnnotationFileNotFoundException,
+  JsonProtocol }
+import firrtl.transforms.{
+  BlackBoxTargetDirAnno,
+  DontCheckCombLoopsAnnotation,
+  NoDCEAnnotation }
+import logger.LogLevel
+import java.io.File
+import net.jcazevedo.moultingyaml._
+import firrtl.annotations.AnnotationYamlProtocol._
+import scala.util.{Try, Failure}
+
+/** Indicates that a subclass is an [[annotations.Annotation]] with an
+  * option consummable by [[HasFirrtlExecutionOptions]]
+  *
+  * This must be mixed into a subclass of [[annotations.Annotation]]
+  */
+trait FirrtlOption { self: Annotation => }
```

Comment:
I think the use of a type here merely for partitioning things below that will then be fed into a massive match statement anyway is a little weird. I think I might be neater if `FirrtlOption` has a virtual method `consume(opt: FirrtlExecutionOptions): FirrtlExecutionOptions ` or something that performs the work in the match below, then just call that method instead of matching.

Reply:
Also would it be possible to seal this trait or make it `private[firrtl]` since users can't just arbitrarily extend this to get something injected into the FirrtlOptions`?